File: lib/mutations/PostTweet.ts
import { graphql, Environment } from "react-relay"
import { commitMutationAsync } from "./commitMutationAsync"

const mutation = graphql`
  mutation PostTweetMutation($input: PostTweetInput!) {
    postTweet(input: $input) {
      tweet {
        id
        body
        mediaURLs
        status
        postedAt
        postedTweetID
      }
    }
  }
`

interface PostTweetInput {
  id: string
  body?: string
  mediaURLs?: readonly string[]
}

export async function postTweet(
  environment: Environment,
  input: PostTweetInput
): Promise<void> {
  const variables = { input }
  await commitMutationAsync(environment, {
    mutation,
    variables,
  })
}

File: ios/Courier/CourierNext/Mutations/PostTweet.swift
import Relay
import RelaySwiftUI

private let mutation = graphql("""
mutation PostTweetMutation($input: PostTweetInput!) {
  postTweet(input: $input) {
    tweetGroup {
      id
      tweets {
        body
        mediaURLs
        postedTweetID
      }
      status
      postedAt
      postAfter
    }
  }
}
""")

extension Mutation.Mutator where Operation == PostTweetMutation {
    func commit(id: String) {
        commit(variables: .init(input: PostTweetInput(id: id)))
    }
}
